spark-indexer/src/test/java/org/uniprot/store/spark/indexer/common/writer/SolrIndexWriterTest.java

```diff
@@ -24,18 +25,55 @@ class SolrIndexWriterTest {
     void canCallWriter() {
         List<SolrInputDocument> docs = new ArrayList<>();
         docs.add(new SolrInputDocument());
+        docs.add(new SolrInputDocument());
+        docs.add(new SolrInputDocument());
         Iterator<SolrInputDocument> iterator = docs.iterator();
         SolrIndexParameter parameter =
                 SolrIndexParameter.builder()
                         .zkHost("zkHost")
                         .collectionName("collectionName")
                         .delay(1L)
                         .maxRetry(1)
+                        .batchSize(2)
                         .build();
         SolrIndexWriter writer = new FakeSolrIndexWriter(parameter, docs, false);
         assertDoesNotThrow(() -> writer.call(iterator));
     }
 
+    @Test
+    void canCallWriterWrongBatchSize() {
+        List<SolrInputDocument> docs = new ArrayList<>();
+        docs.add(new SolrInputDocument());
+        docs.add(new SolrInputDocument());
+        docs.add(new SolrInputDocument());
+        Iterator<SolrInputDocument> iterator = docs.iterator();
+        SolrIndexParameter parameter =
+                SolrIndexParameter.builder()
+                        .zkHost("zkHost")
+                        .collectionName("collectionName")
+                        .delay(1L)
+                        .maxRetry(1)
+                        .batchSize(0)
+                        .build();
+        SolrIndexWriter writer = new FakeSolrIndexWriter(parameter, docs, false);
+        assertThrows(SolrIndexException.class, () -> writer.call(iterator));
+    }
+
+    @Test
+    void canCallWriterEmptyList() {
+        Iterator<SolrInputDocument> iterator = Collections.emptyIterator();
+        SolrIndexParameter parameter =
+                SolrIndexParameter.builder()
+                        .zkHost("zkHost")
+                        .collectionName("collectionName")
+                        .delay(1L)
+                        .maxRetry(1)
+                        .batchSize(1)
+                        .build();
+        SolrIndexWriter writer = new FakeSolrIndexWriter(parameter, new ArrayList<>(), false);
+        assertDoesNotThrow(() -> writer.call(iterator));
+    }
+
     @Test
     void callWriterFail() throws Exception {
         List<SolrInputDocument> docs = new ArrayList<>();
```

Comment:
Good improvement to avoid missing of docs. However, i don't see any integration test with batch size which proves that the docs are finally indexed in multiple iteration.

Reply:
I implemented it. Thank you